build.sh

```diff
@@ -13,16 +13,107 @@
 # the devfiles in the registry will instead use the value in ./VERSION.
 #
 
+#!/bin/bash
+#
+# Copyright (c) 2012-2018 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+set -e
+
+REGISTRY="quay.io"
+ORGANIZATION="eclipse"
+TAG="nightly"
+TARGET="registry"
+DOCKERFILE="./build/dockerfiles/Dockerfile"
+
+USAGE="
+Usage: ./build.sh [OPTIONS]
+Options:
+    --help
+        Print this message.
+    --tag, -t [TAG]
+        Docker image tag to be used for image; default: 'nightly'
+    --registry, -r [REGISTRY]
+        Docker registry to be used for image; default 'quay.io'
+    --organization, -o [ORGANIZATION]
+        Docker image organization to be used for image; default: 'eclipse'
+    --offline
+        Build offline version of registry, with all sample projects
+        cached in the registry; disabled by default.
+    --rhel
+        Build registry using UBI images instead of default
+"
+
+function print_usage() {
+    echo -e "$USAGE"
+}
+
+function parse_arguments() {
+    while [[ $# -gt 0 ]]; do
+        key="$1"
+        case $key in
+            -t|--tag)
+            TAG="$2"
+            shift; shift;
+            ;;
+            -r|--registry)
+            REGISTRY="$2"
+            shift; shift;
+            ;;
+            -o|--organization)
+            ORGANIZATION="$2"
+            shift; shift;
+            ;;
+            --offline)
+            TARGET="offline-registry"
+            shift
+            ;;
+            --rhel)
+            DOCKERFILE="./build/dockerfiles/rhel.Dockerfile"
+            shift
+            ;;
+            *)
+            print_usage
+            exit 0
+        esac
+    done
+}
+
+parse_arguments "$@"
+
+IMAGE="${REGISTRY}/${ORGANIZATION}/che-plugin-registry:${TAG}"
+echo -n "Building image '$IMAGE' "
+if [ "$OFFLINE" = true ]; then
+    echo "in offline mode"
```

Comment:
why not simpler...

```
if [ "$OFFLINE" = true ]; then TARGET="offline-registry"; fi
docker build \
        -t "$IMAGE" \
        -f ./build/dockerfiles/Dockerfile \
        --build-arg LATEST_ONLY="${LATEST_ONLY}" \
        --target ${TARGET} .
...
```

Reply:
Copy paste error from plugin registry build.sh. Fixed.